services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -138,9 +148,18 @@ public void processUnprocessedFederationBatches() {
     return diagnosisKeyBatch.getKeysList()
         .stream()
         .filter(validFederationKeyFilter::isValid)
-        .map(diagnosisKey -> DiagnosisKey.builder().fromFederationDiagnosisKey(diagnosisKey)
-            .withFieldNormalization(new FederationKeyNormalizer(config))
-            .build())
+        .map(diagnosisKey -> {
+          try {
```

Comment:
worth to extract a method for trying to create the DiagnosisKey from the federation key and hide the try catch.

Maybe map(
                 Optional.of(  transformToDiagnosisKey(federationKey) ) 
            ....)